File: DataProcessor.go
package main

import (
	"fmt"
	"sync"
	"time"
)

type Task struct {
	ID int
}

type Result struct {
	TaskID   int
	WorkerID int
	Output   string
}

func worker(id int, tasks <-chan Task, results chan<- Result, wg *sync.WaitGroup) {
	defer wg.Done()
	for task := range tasks {
		fmt.Printf("Worker %d started Task-%d\n", id, task.ID)
		time.Sleep(300 * time.Millisecond) // simulate work
		fmt.Printf("Worker %d completed Task-%d\n", id, task.ID)
		results <- Result{
			TaskID:   task.ID,
			WorkerID: id,
			Output:   fmt.Sprintf("Result of Task-%d by Worker-%d", task.ID, id),
		}
	}
}

func main() {
	numWorkers := 4
	numTasks := 10

	tasks := make(chan Task, numTasks)
	results := make(chan Result, numTasks)

	var wg sync.WaitGroup

	// Start workers
	for w := 0; w < numWorkers; w++ {
		wg.Add(1)
		go worker(w, tasks, results, &wg)
	}

	// Send tasks
	for t := 1; t <= numTasks; t++ {
		tasks <- Task{ID: t}
	}
	close(tasks)

	wg.Wait()
	close(results)

	// Print results
	fmt.Println("\nFinal Results:")
	for result := range results {
		fmt.Println(result.Output)
	}
}

File: DataProcessor.java
import java.io.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.logging.*;

public class DataProcessor {
    private static final int NUM_WORKERS = 4;
    private static final int NUM_TASKS = 10;

    private static final BlockingQueue<String> taskQueue = new LinkedBlockingQueue<>();
    private static final List<String> results = Collections.synchronizedList(new ArrayList<>());
    private static final Logger logger = Logger.getLogger(DataProcessor.class.getName());

    public static void main(String[] args) {
        setupConsoleLogger(); // only console logging

        for (int i = 1; i <= NUM_TASKS; i++) {
            taskQueue.add("Task-" + i);
        }

        ExecutorService executor = Executors.newFixedThreadPool(NUM_WORKERS);
        for (int i = 0; i < NUM_WORKERS; i++) {
            executor.execute(new Worker(i));
        }

        executor.shutdown();
        try {
            executor.awaitTermination(1, TimeUnit.MINUTES);
        } catch (InterruptedException e) {
            logger.log(Level.SEVERE, "Executor interrupted", e);
        }

        writeResults(); // fallback to console if file fails
    }

    private static void setupConsoleLogger() {
        logger.setUseParentHandlers(false); // disable default
        ConsoleHandler handler = new ConsoleHandler();
        handler.setFormatter(new SimpleFormatter());
        logger.addHandler(handler);
    }

    private static void writeResults() {
        try (PrintWriter out = new PrintWriter("results_java.txt")) {
            for (String result : results) {
                out.println(result);
            }
            logger.info("Results written to results_java.txt");
        } catch (IOException e) {
            logger.severe("Could not write to file. Printing results instead:");
            for (String result : results) {
                System.out.println(result);
            }
        }
    }

    static class Worker implements Runnable {
        private final int workerId;

        Worker(int id) {
            this.workerId = id;
        }

        @Override
        public void run() {
            while (true) {
                try {
                    String task = taskQueue.poll(1, TimeUnit.SECONDS);
                    if (task == null) break;

                    logger.info("Worker " + workerId + " started " + task);
                    processTask(task);
                    logger.info("Worker " + workerId + " completed " + task);
                } catch (InterruptedException e) {
                    Thread.currentThread().interrupt();
                    logger.warning("Worker " + workerId + " interrupted");
                    break;
                } catch (Exception e) {
                    logger.severe("Worker " + workerId + " error: " + e.getMessage());
                }
            }
        }

        private void processTask(String task) throws InterruptedException {
            Thread.sleep(500);
            results.add("Result of " + task + " by Worker-" + workerId);
        }
    }
}
